codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/GoIntegration.java

```diff
@@ -186,4 +187,34 @@ default List<RuntimeClientPlugin> getClientPlugins() {
     default String processServiceId(GoSettings settings, Model model, String serviceId) {
         return serviceId;
     }
+
+    /**
+     * Used by integrations to set a member on the endpoints resolution
+     * middleware object.
+     *
+     * @param writer Settings used to generate.
+     */
+    default void renderEndpointBuiltInField(GoWriter writer) {
```

Comment:
**suggestion**: It might be better to collapse these into one method that returns a type responsible for these different things. 

e.g.

```
default void endpointBuiltinHandler(): EndpointBuiltInHandler


interface EndpointBuiltInHandler {
    default void renderEndpointBuiltInField(GoWriter writer) { }

    default void renderEndpointBuiltInInvocation(GoWriter writer) {}

    default void renderEndpointBuiltInInitialization(GoWriter writer, Parameters parameters) { }
}
```

Grouping these things together reduces the API surface of `GoIntegration`, it also helps in finding all the various methods that may need overridden for related functionality.